pyomo/core/base/indexed_component_slice.py

```diff
@@ -263,11 +264,13 @@ def __str__(self):
         for level in self._call_stack:
             if level[0] == IndexedComponent_slice.slice_info:
                 ans += level[1][0].name
-                tmp = dict(level[1][1])
-                tmp.update(level[1][2])
-                if level[1][3] is not None:
-                    tmp[level[1][3]] = Ellipsis
-                ans += self._getitem_args_to_str([tmp[i] for i in sorted(tmp)])
+                if level[1][1] is not None:
+                    tmp = dict(level[1][1])
+                    tmp.update(level[1][2])
+                    if level[1][3] is not None:
+                        tmp[level[1][3]] = Ellipsis
+                    ans += self._getitem_args_to_str(
+                        [tmp[i] for i in sorted(tmp)])
```

Comment:
Do you still need `sorted` here?

Reply:
Yes - I think so: that is putting things back in argument order (the tmp dict maps index position to the fixed / wildcard / ellipsis value)